src/Internal/Cardano/Types/Transaction.purs

```diff
@@ -788,26 +814,52 @@ derive newtype instance EncodeAeson Vkey
 instance Show Vkey where
   show = genericShow
 
-newtype PublicKey = PublicKey Bech32String
+newtype PublicKey = PublicKey RawBytes
+
+mkPubKey :: Bech32String -> Maybe PublicKey
+mkPubKey = map (PublicKey <<< bytesFromPublicKey) <<< publicKeyFromBech32
+
+mkFromCslPubKey :: Serialization.PublicKey -> PublicKey
+mkFromCslPubKey = PublicKey <<< bytesFromPublicKey
+
+convertPubKey :: PublicKey -> Serialization.PublicKey
+convertPubKey (PublicKey bs) = unsafePartial $ fromJust <<< fromBytes <<< unwrap
+  $ bs
 
-derive instance Generic PublicKey _
-derive instance Newtype PublicKey _
 derive newtype instance Eq PublicKey
 derive newtype instance Ord PublicKey
 derive newtype instance EncodeAeson PublicKey
 
+instance ToData PublicKey where
+  toData (PublicKey bytes) = toData bytes
+
+instance FromData PublicKey where
+  fromData = map mkFromCslPubKey <<< fromBytes <=< fromData
+
 instance Show PublicKey where
-  show = genericShow
+  show pk = "PublicKey " <> (bech32FromPublicKey <<< convertPubKey $ pk)
```

Comment:
You'll need to add some parens around the string, otherwise this will evaluate to a string representing invalid Purescript code in some cases
```suggestion
  show pk = "(PublicKey " <> bech32FromPublicKey (convertPubKey pk) <> ")"
```